### PR TITLE
deps!: use cosmos-sdk v0.45.16-lsm-ics

### LINKED ISSUE #2682: integrate ICS with LSM changes
Gaia needs to use a version of ICS v2 that supports the LSM module in order to fully support the feature without any impact on Gaia's livenes.

The changes are state breaking for Gaia and warrant a governance vote to upgrade.

----
## Proposed timeline:

After LSM changes are introduced to `cosmos-sdk` make sure to bump ICS to the correct version that supports the LSM functionality.

ICS changes will likely land into version `v2.1`

The issue tracking changes to ICS: https://github.com/cosmos/interchain-security/issues/1183
* PR: https://github.com/cosmos/interchain-security/pull/1120

# Instructions
When the [ICS PR integrating LSM](https://github.com/cosmos/interchain-security/pull/1120) is merged and ICS v2.1 is cut, refactor gaia to use the latest supported versions of:
* cosmos-sdk (v0.45.16-lsm-ics)
* interchain-securirt (v2.1.0)

### PR DESCRIPTION
**Pull request by MSalopek**
## Description

Closes: #2682 

Dependency bump and tests for [cosmos-sdk v0.45.16-lsm-ics integration](https://github.com/cosmos/cosmos-sdk/pull/16747) with gaia .

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Provided a link to the relevant issue or specification
- [ ] Reviewed "Files changed" and left comments if necessary 
- [ ] Confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] Confirmed all author checklist items have been addressed
- [ ] Confirmed that this PR does not change production code

### CODE REVIEW

**Comment by faddat** [comment]
Hey sir, I just want to make sure that the dots get connected here, there are some fixes here:

* https://github.com/iqlusioninc/liquidity-staking-module/pull/117
* https://github.com/iqlusioninc/liquidity-staking-module/pull/116
* https://github.com/iqlusioninc/liquidity-staking-module/pull/112

**Comment by xlab** [comment]
@faddat thanks for keeping track of it, as far as I understand these are incorporated already in the cosmos-sdk [v0.45.16-ics-lsm](https://github.com/cosmos/cosmos-sdk/compare/v0.45.16-ics-lsm...iqlusioninc:cosmos-sdk:v0.45.16-ics-lsm) PR

File: app/upgrades/v12/upgrades.go [hunk 1]
```diff
@@ -21,6 +21,14 @@ func CreateUpgradeHandler(
...
 		}
 
+		// Set liquid staking module parameters
+		params := keepers.StakingKeeper.GetParams(ctx)
+		params.ValidatorBondFactor = sdk.NewDec(ValidatorBondFactor)
+		params.ValidatorLiquidStakingCap = sdk.NewDec(ValidatorLiquidStakingCap)
+		params.GlobalLiquidStakingCap = sdk.NewDec(GlobalLiquidStakingCap)
+
+		keepers.StakingKeeper.SetParams(ctx, params)
+
```
Comment (mpoke) [inline app/upgrades/v12/upgrades.go:31]: Why are you saying that? `RunMigrations` should bump the staking ConsensusVersion to 3 (https://github.com/cosmos/cosmos-sdk/blob/201528c4c4c1661a4b7e3a5510f40fb6865df689/x/staking/module.go#L168) and execute the [Migrate2to3 method](https://github.com/cosmos/cosmos-sdk/blob/201528c4c4c1661a4b7e3a5510f40fb6865df689/x/staking/keeper/migrations.go#L25). This call will add the LSM params (by calling [MigrateParamsStore](https://github.com/cosmos/cosmos-sdk/blob/201528c4c4c1661a4b7e3a5510f40fb6865df689/x/staking/migrations/v3/store.go#L36C6-L36C24)). Then we change the default values of these params in the upgrade handler.

Comment (mpoke) [reply app/upgrades/v12/upgrades.go:31]: @riley-stride By the way, shouldn't we call `RefreshTotalLiquidStaked` after updating the param values? At the moment it's called by the migration code code in the SDK, see https://github.com/cosmos/cosmos-sdk/blob/201528c4c4c1661a4b7e3a5510f40fb6865df689/x/staking/migrations/v3/store.go#L143

**Comment by yaruwangway** [comment]
Hi, i get err in upgrade:
```
Staking LSM Migration: Calculating total liquid staked
panic: value from ParamSetPair is invalid: global liquid staking cap cannot be greater than 100%: 25.000000000000000000
```
and i cannot set the `GlobalLiquidStakingCap ` = 0.25 rather than 25 because this is int type.
the same for `ValidatorLiquidStakingCap`

File: app/upgrades/v12/constants.go [hunk 2]
```diff
@@ -7,6 +7,17 @@ import (
 const (
 	// UpgradeName defines the on-chain upgrade name.
 	UpgradeName = "v12"
+
+	// The ValidatorBondFactor dictates the cap on the liquid shares
+	// for a validator - determined as a multiple to their validator bond
+	// (e.g. ValidatorBondShares = 1000, BondFactor = 250 -> LiquidSharesCap: 250,000)
+	ValidatorBondFactor = 250
+	// GlobalLiquidStakingCap represents the percentage cap on
```
Comment (shaspitz) [inline app/upgrades/v12/constants.go]: Incorrect comment

File: app/upgrades/v12/constants.go [hunk 3]
```diff
@@ -7,6 +7,17 @@ import (
...
+	// the portion of a validator's stake that can be liquid
+	ValidatorLiquidStakingCap = 50 // 50%
```
Comment (shaspitz) [inline app/upgrades/v12/constants.go]: This should not be 50 as int, rather 0.5 as `sdk.Dec`, see https://github.com/cosmos/cosmos-sdk/blob/201528c4c4c1661a4b7e3a5510f40fb6865df689/x/staking/types/params.go#L276

File: app/upgrades/v12/constants.go [hunk 4]
```diff
@@ -7,6 +7,17 @@ import (
...
+	// GlobalLiquidStakingCap represents the percentage cap on
+	// the portion of a chain's total stake can be liquid
+	GlobalLiquidStakingCap = 25 // 25%
```
Comment (shaspitz) [inline app/upgrades/v12/constants.go]: Ditto

**Comment by sampocs** [comment]
> Hi, i get err in upgrade:
> 
> ```
> Staking LSM Migration: Calculating total liquid staked
> panic: value from ParamSetPair is invalid: global liquid staking cap cannot be greater than 100%: 25.000000000000000000
> ```
> 
> and i cannot set the `GlobalLiquidStakingCap ` = 0.25 rather than 25 because this is int type. the same for `ValidatorLiquidStakingCap`

Where are you seeing that it's an int? Should be a dec and the value passed should be 0.25 https://github.com/cosmos/cosmos-sdk/blob/feature/v0.45.x-ics-lsm/x/staking/types/params.go#L73-L75

**Comment by shaspitz** [comment]
> Where are you seeing that it's an int? Should be a dec and the value passed should be 0.25 https://github.com/cosmos/cosmos-sdk/blob/feature/v0.45.x-ics-lsm/x/staking/types/params.go#L73-L75

See [here](https://github.com/cosmos/gaia/blob/be57325f85a2d17298872ce4076bad503bdefedd/app/upgrades/v12/constants.go)

**Comment by sampocs** [comment]
> > Where are you seeing that it's an int? Should be a dec and the value passed should be 0.25 https://github.com/cosmos/cosmos-sdk/blob/feature/v0.45.x-ics-lsm/x/staking/types/params.go#L73-L75
> 
> See [here](https://github.com/cosmos/gaia/blob/be57325f85a2d17298872ce4076bad503bdefedd/app/upgrades/v12/constants.go)

Those are just constants, they're cast as decimals when they're actually set [here](https://github.com/cosmos/gaia/blob/be57325f85a2d17298872ce4076bad503bdefedd/app/upgrades/v12/upgrades.go#L26-L28). 

We could change them to `vars` and define them as decimals if you'd prefer

**Comment by sampocs** [comment]
jinx @smarshall-spitzbart

**Comment by shaspitz** [comment]
Haha yea they were prev being cast incorrectly, but things should be good after https://github.com/cosmos/gaia/pull/2643/commits/6a99e8da70a7e71aea376354082afcd3f4916da9

File: app/upgrades/v12/upgrades.go (hunk 1 above)
Comment (sampocs) [reply app/upgrades/v12/upgrades.go:31]: `RefreshTotalLiquidStaked` doesn't reference the params. It just sets the `TotalLiquidStakedTokens` and `LiquidShares` on each validator

**Comment by yaruwangway** [comment]
sry, not int @smarshall-spitzbart  fixed it. 6a99e8da70a7e71aea376354082afcd3f4916da9

**Review by mpoke — APPROVED** [review]

**Review by shaspitz — APPROVED** [review]

**Pull request merged**